plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -505,86 +499,112 @@ public void pull(final PullParams params, final ProgressMonitor progressMonitor)
     /**
      * Creates an ImageStreamTag that tracks a given image.
      *
-     * <p>
-     * The target is usually specified e.g. eclipse-che/workspace<unique string>.
-     * ImageStreamTags are labelled as {@code <registry>/<namespace>/<ImageStream>:<tag>}
-     * The ImageStream name must be unique to the source image (e.g. eclipse_ubuntu_jdk8)
-     * otherwise the sourceImage for this ImageStream will be overwritten when
-     * another Stack is used. This leaves only the tag to hold the new image name.
+     * <p> Docker tags are used extensively in Che: all workspaces run on tagged images
+     * tracking built stacks. For new workspaces, or when snapshots are not used, the
+     * tracked image is e.g. {@code eclipse/ubuntu_jdk8}, whereas for snapshotted workspaces,
+     * the tracked image is the snapshot (e.g. {@code machine_snapshot-<identifier>}.
+     *
+     * <p> Since OpenShift does not support the same tagging functionality as Docker,
+     * tags are implemented as ImageStreamTags, where the {@code From} field is always
+     * the original image, and the ImageStreamTag name is derived from both the source
+     * image and the target image. This replicates functionality for Che in Docker,
+     * while working differently under the hood. The ImageStream name is derived from
+     * the image that is being tracked (e.g. {@code eclipse/ubuntu_jdk8}), while the tag
+     * name is derived from the target image (e.g. {@code eclipse-che/che_workspace<identifier>}).
      *
      * @see DockerConnector#tag(TagParams)
      */
     @Override
     public void tag(final TagParams params) throws ImageNotFoundException, IOException {
-        // E.g. `docker tag oldImageRepo newImageRepo`
+        // E.g. `docker tag sourceImage targetImage`
         String sourceImage = params.getImage();        // e.g. eclipse/ubuntu_jdk8
         String targetImage = params.getRepository();   // e.g. eclipse-che/<identifier>
         String tag = params.getTag();
-        if (tag == null || tag.isEmpty()) {
+        if (sourceImage.contains(":")) {
```

Comment:
You split images string into tag and image name multiple times in the code. To avoid duplication you can probably extract that logic in a new method.